File: Programming - 2/Lab_5_6/Lab_6_CPP/Back/Circle.cpp
#include "Circle.h"

Circle::Circle(double r)
{
    if (r <= 0)
        throw "Radius must be positive";

    _radius = r;
}

Circle::Circle(double r, Point2D c) : Circle::Circle(r)
{
    _center = c;
}

double Circle::GetRadius() { return _radius; }
Point2D Circle::GetCenter() { return _center; }

double Circle::GetPerimeter() { return 2 * 3.14 * _radius; }
double Circle::GetArea() { return 3.14 * _radius * _radius; }

File: Programming - 2/Lab_5_6/Shapes/Circle.cs
using System;

namespace Shapes
{
    public sealed class Circle : Shape
    {
        public Circle(double r, (double x, double y) c)
        {
            if (r <=0)
            {
                throw new ArgumentException("Radius must be positive", "r");
            }
            Radius = r;
            Center = c;
        }

        public double Radius { get; private set; }
        public (double x, double y) Center { get; private set; }
        public override double Area => Math.PI * Radius * Radius;
        public override double Perimeter => 2  * Math.PI * Radius;
    }
}
